src/cryptsetup/cryptsetup-tokens/luks2-tpm2.c

```diff
@@ -75,6 +76,14 @@ int acquire_luks2_key(
                 r = tpm2_pcrlock_policy_load(pcrlock_path, &pcrlock_policy);
                 if (r < 0)
                         return r;
+                if (r == 0) {
+                        /* Not found? Then search among passed credentials */
+                        r = tpm2_pcrlock_policy_from_credentials(srk, pcrlock_nv, &pcrlock_policy);
+                        if (r < 0)
+                                return r;
+                        if (r == 0)
+                                return log_error_errno(SYNTHETIC_ERRNO(EREMOTE), "Couldn't find pcrlock policy for volume.");
```

Comment:
Shouldn't this be lower than error?

Reply:
no, it should be an error. if the enrolled dat asays we need a pcrlock policy and we can#t find it, we must fail the unlock operation, we cannot possibly fulfill the policy.

There was a bug here actually before this change, since tpm2_pcrlock_policy_load() above would return 0 if we didn't find a policy already before, and we didn't turn that into a failure here. This rectifies this, and makes sure we correctly give up if we really cannot find any policy.